pkg/util/apigroup_test.go

```diff
@@ -240,3 +241,134 @@ func TestDefaultSkippedResourceConfigGroupVersionKindParse(t *testing.T) {
+		input           string
+		expectedErr     error
+		expectedGroups  []string
+		expectedGroupVs []schema.GroupVersion
+		expectedGvks    []schema.GroupVersionKind
+	}{
+		{
+			input:           "",
+			expectedErr:     nil,
+			expectedGroups:  []string{},
+			expectedGroupVs: []schema.GroupVersion{},
+			expectedGvks:    []schema.GroupVersionKind{},
+		},
+		{
+			input:           "apps",
+			expectedErr:     nil,
+			expectedGroups:  []string{"apps"},
+			expectedGroupVs: []schema.GroupVersion{},
+			expectedGvks:    []schema.GroupVersionKind{{Group: "apps"}},
+		},
+		{
+			input:           "apps/v1",
+			expectedErr:     nil,
+			expectedGroups:  []string{},
+			expectedGroupVs: []schema.GroupVersion{{Group: "apps", Version: "v1"}},
+			expectedGvks:    []schema.GroupVersionKind{},
+		},
+		{
+			input:           "apps/v1beta1",
+			expectedErr:     nil,
+			expectedGroups:  []string{},
+			expectedGroupVs: []schema.GroupVersion{{Group: "apps", Version: "v1beta1"}},
+			expectedGvks:    []schema.GroupVersionKind{},
+		},
+		{
+			input:           "apps/v1/Deployment",
+			expectedErr:     nil,
+			expectedGroups:  []string{},
+			expectedGroupVs: []schema.GroupVersion{},
+			expectedGvks:    []schema.GroupVersionKind{{Group: "apps", Version: "v1", Kind: "Deployment"}},
+		},
+		{
+			input:           "apps/v1beta1/Deployment",
+			expectedErr:     nil,
+			expectedGroups:  []string{},
+			expectedGroupVs: []schema.GroupVersion{},
+			expectedGvks:    []schema.GroupVersionKind{{Group: "apps", Version: "v1beta1", Kind: "Deployment"}},
+		},
+		{
+			input:           "apps/v1/Deployment,ReplicaSet",
+			expectedErr:     nil,
+			expectedGroups:  []string{},
+			expectedGroupVs: []schema.GroupVersion{},
+			expectedGvks: []schema.GroupVersionKind{
+				{Group: "apps", Version: "v1", Kind: "Deployment"},
+				{Group: "apps", Version: "v1", Kind: "ReplicaSet"},
+			},
+		},
+		{
+			input:           "v1/Service",
+			expectedErr:     nil,
+			expectedGroups:  []string{},
+			expectedGroupVs: []schema.GroupVersion{},
+			expectedGvks:    []schema.GroupVersionKind{{Group: "", Version: "v1", Kind: "Service"}},
+		},
+		{
+			input:           "v1/Service,ConfigMap",
+			expectedErr:     nil,
+			expectedGroups:  []string{},
+			expectedGroupVs: []schema.GroupVersion{},
+			expectedGvks: []schema.GroupVersionKind{
+				{Group: "", Version: "v1", Kind: "Service"},
+				{Group: "", Version: "v1", Kind: "ConfigMap"},
+			},
+		}, {
+			input:           "///",
+			expectedErr:     fmt.Errorf("invalid parameter: ///"),
+			expectedGroups:  []string{},
+			expectedGroupVs: []schema.GroupVersion{},
+			expectedGvks:    []schema.GroupVersionKind{},
+		},
+	}
+
+	for i, test := range tests {
+		t.Run(fmt.Sprintf("Test case %d", i+1), func(t *testing.T) {
+			// Create a new SkippedResourceConfig
+			r := NewSkippedResourceConfig()
+
+			// Call parseSingle with the input
+			err := r.parseSingle(test.input)
+
+			if !isErrorMsgEqual(err, test.expectedErr) {
```

Comment:
Check whether the err exists will be ok, just like other test case.

Reply:
I designed a situation for it that might make mistakes...

Reply:
> I designed a situation for it that might make mistakes...

What?

I agree with @XiShanYongYe-Chang that a `boolean` flag should be ok since the `parseSingle` only returns one error, so we don't need to check if it is the expected one.